src/content/blog/2024-08-27-integration-tests.mdx

```diff
@@ -0,0 +1,130 @@
+
+![Integration Test Architecture Diagram](/developers/img/blog/2024-08-27/architecture.png)
+
+#### Launching the Test Environment and Running Tests
+
+The environment and tests are launched from a shell script that does the following:
+
+- Starts the test environment consisting of our two Rafiki instances from our Docker Compose files.
+- Maps hosts to our host machine’s hosts file.
+- Runs the Jest test suite against our Rafiki instances in Docker and saves the test exit code.
+- Tears down the Docker environment and returns the test exit code.
+
+#### Test Platform Components
+
+Using Jest as our testing framework, we structured our test code around a few key components:
+
+1. **Mock Account Servicing Entity (MASE)**
+
+   - **Integration Server**: Includes all endpoints needed for integrating Account Servicing Entities with Rafiki. This includes the [rates endpoint](https://rafiki.dev/integration/getting-started/#exchange-rates) for supplying currency exchange rates and an endpoint for handling Rafiki’s [webhook events](https://rafiki.dev/integration/webhook-events/) throughout the payment lifecycle, such as depositing liquidity on `outgoing_payment.created`.
+   - **Open Payments Client**: Communicates with our Open Payments API to perform the payment flows. This API is a reference implementation of the [Open Payments standard](https://rafiki.dev/concepts/open-payments/overview/) that enables third parties to directly access users’ accounts.
+   - **Admin (GraphQL) Client**: Communicates with our GraphQL admin API to set up tests and complete some of the flows, such as “Peer-to-Peer” payments.
+   - **Account Provider**: A simple accounting service to handle basic accounting functions and facilitate payment flows.
+
+2. **Test Actions:** These are functions analogous to our Bruno requests but designed to be repeatable across tests. These actions abstract away some baseline assumptions about the sending/receiving MASE relationship, assertions, and how each endpoint is called.
+
+On test start, we create MASEs for `cloud-nine-wallet` and `happy-life-bank` and seed their respective Rafiki instances. Then these are designated as `sendingASE` and `receivingASE` in our test actions and we run our test flows with several variations. These include:
+
+**Open Payments:**
+
+- With Continuation via Polling
+- With Continuation via Finish method
+- Without Quote step
+
+**Peer-to-Peer:**
+
+- Single currency
+- Cross currency
+
+#### URL Handling
```

Comment:
I plan on adding a bit more here - just stating how it was added to our github action and saying how this constitutes a much more complete/consistent test of our changes. just wrapping it up similarly to how i will in the conclusion basically.

Reply:
Updated with how it was added to CI and an explanation for both open payments and p2p. 

I kept the description fairly brief and refered to our open payments docs for more details on the flow in general and the grant aspect. I wanted to avoid opening a can of worms that would not be easy to resolve here ("whats an incoming payment", "why create resources in that order", "what are the grants about?" etc.). I think the links take care of it though and its a good segue into the documentation.

I characterized the OP flow as something that would be use in an ecommerce context... wanted to make it more relatable but not sure if that is too suggestive (could be web mon, maybe card payment, etc.).